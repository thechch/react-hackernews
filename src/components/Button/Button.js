import React from 'react';
import PropTypes from 'prop-types';

import { Loading } from '../Loading';
import './Button.css';

const Button = ({ onClick, className, children }) => (
  <button onClick={onClick} className={className} type="button">
    {children}
  </button>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.string,
};

Button.defaultProps = {
  className: '',
  children: '',
};

const withLoading = (Component) => {
  return function WrappedWithLoading({ isLoading, ...rest }) {
    return isLoading ? <Loading {...rest} /> : <Component {...rest} />;
  };
};

const ButtonWithLoading = withLoading(Button);

export default ButtonWithLoading;
