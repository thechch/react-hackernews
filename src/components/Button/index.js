export { Button, ButtonWithLoading } from './Button';
