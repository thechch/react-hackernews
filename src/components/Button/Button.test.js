import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  const props = {
    onClick: () => document.createElement('test'),
  };
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button {...props}>Give me more</Button>, div);
  });

  test('snapshot', () => {
    const component = renderer.create(<Button {...props}>Give me more</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows two items in list', () => {
    const element = shallow(<Button {...props} />);
    element.find('button').simulate('click');
    expect(element.contains(<input />));
  });
});
