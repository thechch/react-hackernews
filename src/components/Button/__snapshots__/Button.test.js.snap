exports[`Button snapshot 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button">
  Give me more
</button>
`;
